### ISSUE TITLE
Add Guard for SmartEnums out of range.

### ISSUE THREAD

**Issue by MarkLFT** [ardalis/GuardClauses#346]
It would be good to have an OutOfRange for SmartEnums the same as regular Enums.

**Comment by ardalis** [comment]
Agreed. Is it worth its own separate package though? I wouldn't want to force a dependency on SmartEnum into GuardClauses so it would have to exist in a separate (very small) Ardalis.GuardClauses.SmartEnum package.

Or I could just add guards to SmartEnum itself which might make more sense. Thoughts?

**Comment by ardalis** [comment]
Right, so this would argue for me adding them to SmartEnum project, not this one? In that case, would you mind creating an issue over there and reference this one?

**Comment by dudi-w** [comment]
Can this issue be closed?
